File: src/main/java/fr/lewon/web/bot/entities/input/champions/ActionChampionsTeamDraft.java
package fr.lewon.web.bot.entities.input.champions;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import fr.lewon.bot.http.body.urlencoded.FUEMember;

public class ActionChampionsTeamDraft extends ActionChampions {

	@FUEMember("id_champion")
	private Integer idChampion;
	
	@FUEMember
	private String namespace = "h%5CChampions";
	
	@FUEMember(value = "girls_to_keep%5B%5D")
	private List<Integer> girlsToKeep;
	
	public ActionChampionsTeamDraft(Integer idChampion, Integer... girlsToKeep) {
		this(idChampion, Arrays.asList(girlsToKeep));
	}
	
	public ActionChampionsTeamDraft(Integer idChampion, List<Integer> girlsToKeep) {
		super("team_draft");
		this.idChampion = idChampion;
		this.girlsToKeep = new ArrayList<>(girlsToKeep);
	}

}

File: src/main/kotlin/fr/lewon/bot/hh/entities/input/champions/ActionChampionsTeamDraft.kt
package fr.lewon.bot.hh.entities.input.champions

import com.fasterxml.jackson.annotation.JsonProperty


class ActionChampionsTeamDraft(idChampion: Int, girlsToKeep: List<Int>) : ActionChampions("team_draft") {
    @JsonProperty(value = "id_champion")
    val idChampion: Int = idChampion
    @JsonProperty
    val namespace = "h\\Champions"
    @JsonProperty(value = "girls_to_keep[]")
    val girlsToKeep: List<Int> = girlsToKeep

}